### DISCUSSION TITLE
How do I add metadata to my library.

### DISCUSSION THREAD

**Discussion by tfyt2023** [stashapp/stash#5574] [Q&A]
Trying to figure out how to auto-tag performers, scenes, and any other meta-data that this app can.

When I go to Settings > Metadata providers I see the following message

"Stash-box facilitates automated tagging of scenes and performers based on fingerprints and filenames. Endpoint and API key can be found on your account page on the stash-box instance. Names are required when more than one instance is added."

But where is the "account page on the stash-box instance".... I don't see the word "account" anywhere in the app. Kind of frustrating that instructions are pointing to places that seemingly do not exist.

Also, I went to Tasks > Auto-tag > then clicked the Auto-tag button. It said it was tagging my videos, yet when it was done nothing had any tags.

So how exactly do you auto tag your collection? A step by step instruction would be nice. This is honestly very confusing, and not user friendly at all.

**Comment by DogmaDragon** [comment]
> But where is the "account page on the stash-box instance".... I don't see the word "account" anywhere in the app. Kind of frustrating that instructions are pointing to places that seemingly do not exist.

https://docs.stashapp.cc/metadata-sources/stash-box-instances

> Also, I went to Tasks > Auto-tag > then clicked the Auto-tag button. It said it was tagging my videos, yet when it was done nothing had any tags.

Auto Tag task only works with already created objects, it doesn't create them itself. https://docs.stashapp.cc/in-app-manual/tasks/autotagging/

**Comment by tfyt2023** [comment]
>https://docs.stashapp.cc/metadata-sources/stash-box-instances

I signed up for StashDB and got the "GraphQL endpoint" and API key....

I went to the Settings > Metadata Providers and used that info to add a "stashbox endpoint".

Auto-tag still doesn't do anything when I click it.

>Auto Tag task only works with already created objects, it doesn't create them itself.

I don't understand what you mean by "already created objects".... What is an "object".... How do I create one? Etc... I already have a bunch of videos added to the stash-app, but I'm assuming that's not what you're referring to.

Sorry if I'm coming off as stupid, but I don't feel like I'm any closer to actually understanding how to auto-tag stuff.

Can you please just give me a simple rundown on what I'm supposed to do in the app? Like... assume you're talking to a monkey, and tell me where to click and what to do if you can.

>https://docs.stashapp.cc/in-app-manual/tasks/autotagging/

I already read that page before posting my original comment and still couldn't figure out what to do. I appreciate you taking the time to try and help, but feel like I'm going down a rabbit hole of documentation leading to other documentation, leading to more documentation. And it's all using jargon I don't understand, and not containing practical instructions, only abstractions.

I really need just a set of practical instructions for an end-user. I've got a bunch of videos imported into stash, but the only info that appears is the filename, and I have no clue what I'm supposed to do from there to actually organize and tag this content.
